### ISSUE TITLE
bufferCount seems broken

### ISSUE THREAD

**Issue by VerTiGoEtrex** [ReactiveX/rxdart#206]
I wrote a simple test to show the behavior I would expect from bufferCount, and it does not pass. Been trying to wrap my head around the implementation, but buffer transformers are not exactly straightforward. In particular, it seems the "skip" behavior, specifically, is broken, or I have some misunderstanding of how skip should work.

Here is my test:
```dart
test('Ensure bufferCount acts as I expect', () async {
    final observable = Observable.range(0, 7)
    .bufferCount(5, 2);
    
    await expectLater(observable, emitsInOrder([
      [0,1,2,3,4],
      [2,3,4,5,6],
      [4,5,6,7],
    ]));
  });
```

If this is not the expected output, then the "KonamiCode" example on the readme is sorely misleading. For example, this test does not pass either (it "types" 'B' followed by the KonamiCode)

```dart
enum KeyCode {
  UP, DOWN, LEFT, RIGHT, B, A
}

test('konami', () async {
    const konamiKeyCodes = const <KeyCode>[
      KeyCode.UP,
      KeyCode.UP,
      KeyCode.DOWN,
      KeyCode.DOWN,
      KeyCode.LEFT,
      KeyCode.RIGHT,
      KeyCode.LEFT,
      KeyCode.RIGHT,
      KeyCode.B,
      KeyCode.A];

    final testKeyCodes = <KeyCode>[
      KeyCode.B,
    ]..addAll(konamiKeyCodes);

    final obs = Observable.fromIterable(testKeyCodes)
        .bufferCount(10, 1)
        .where((lastTenKeyCodes) => const IterableEquality<KeyCode>().equals(lastTenKeyCodes, konamiKeyCodes));

    await expectLater(obs, emits(konamiKeyCodes));
  });
```
It emits
```
   Which: emitted • [KeyCode.B, KeyCode.UP, KeyCode.UP, KeyCode.DOWN, KeyCode.DOWN, KeyCode.LEFT, KeyCode.RIGHT, KeyCode.LEFT, KeyCode.RIGHT, KeyCode.B]
                  • [KeyCode.B, KeyCode.A]
                  • [KeyCode.A]
                  x Stream closed.
```

**Comment by brianegan** [comment]
Hey there -- Good call. I don't use skip as often and hadn't noticed, but I played around with RxJava and RxJS implementations, and ours is definitely different. 

@frankpepermans: you know a bit more about this operator. It seems like "skip" should work a bit differently, and in RxJS they even call it `startBufferEvery` which I think is a better / more descriptive name.

Overall, it feels like it should start buffering events every "X" skip events, rather than doing what it's doing now. This means that if you use `bufferCount(2)` and `bufferCount(2, 2)`, they'll produce the exact same output.

E.g. `Observable.range(0, 5).bufferCount(2)` and `Observable.range(0, 5).bufferCount(2, 2)`. should produce:

```
[0, 1]
[2, 3]
[4, 5]
```

However, this is where it gets pretty weird. `bufferCount(2, 2)` produced the following output:

```
[0, 1]
[0, 1, 2, 3]
[2, 3, 4, 5]
[4, 5]
```

**Comment by frankpepermans** [comment]
Yeah I'll take a look, buffer with count is actually based on rxjs 1 from a
couple of years ago.

Might need some love indeed :)

**Comment by frankpepermans** [comment]
see https://github.com/ReactiveX/rxdart/pull/210 @brianegan

**Comment by frankpepermans** [comment]
fixed in 0.20.0
